src/app/components/wrapper-components/part-creation/part-creation.component.ts

```diff
@@ -294,112 +310,130 @@ export class PartCreationComponent implements OnInit, OnDestroy {
+            await this.userService.sendPresenceToken();
+            this.selfTokenInterval = MGPOptional.of(window.setInterval(() => {
+                // eslint-disable-next-line @typescript-eslint/no-floating-promises
+                this.userService.sendPresenceToken();
+            }, PartCreationComponent.TOKEN_INTERVAL));
+        }
+    }
+    public stopSendingPresenceTokensAndObservingUsersIfNeeded(): void {
+        if (this.selfTokenInterval.isPresent()) {
+            window.clearInterval(this.selfTokenInterval.get());
+            this.selfTokenInterval = MGPOptional.empty();
+        }
+        for (const userId of this.usersSubscriptions.listKeys()) {
+            this.unsubscribeFrom(userId);
+        }
+    }
+    private removeUserFromLobby(userId: string, userWasDeleted: boolean = false): Promise<void> {
         const joiner: Joiner = Utils.getNonNullable(this.currentJoiner);
-        const index: number = joiner.candidates.indexOf(username);
+        const index: number = joiner.candidates.findIndex((minimalUser: MinimalUser) => minimalUser.id === userId);
         // The user must be in the lobby, otherwise we would have unsubscribed from its updates
         assert(index !== -1, 'PartCreationComponent: attempting to remove a user not in the lobby');
-        const beforeUser: string[] = joiner.candidates.slice(0, index);
-        const afterUser: string[] = joiner.candidates.slice(index + 1);
-        const candidates: string[] = beforeUser.concat(afterUser);
-        if (username === joiner.chosenPlayer) {
-            // The chosen player has been removed, the user will have to review the config
-            this.messageDisplayer.infoMessage($localize`${username} left the game, please pick another opponent.`);
-            return this.joinerService.reviewConfigRemoveChosenPlayerAndUpdateCandidates(candidates);
-        } else {
-            return this.joinerService.updateCandidates(candidates);
-        }
+        const beforeUser: MinimalUser[] = joiner.candidates.slice(0, index);
+        const afterUser: MinimalUser[] = joiner.candidates.slice(index + 1);
+        const candidates: MinimalUser[] = beforeUser.concat(afterUser);
+        assert(userId === joiner.chosenPlayer || userWasDeleted, 'Should only remove chosenUser from creation room, since candidate are not polled and should remove themselves');
+        // The chosen player has been removed, the user will have to review the config
+        this.messageDisplayer.infoMessage($localize`${userId} left the game, please pick another opponent.`);
+        return this.joinerService.reviewConfigRemoveChosenPlayerAndUpdateCandidates(candidates);
     }
-    private unsubscribeFrom(username: string): void {
-        const subscription: () => void = this.candidateSubscription.delete(username);
+    public unsubscribeFrom(userId: string): void {
+        const subscription: () => void = this.usersSubscriptions.get(userId).get();
```

Comment:
faut-il pas supprimer la subscription en question aussi ?

Reply:
ood ! (out of date :D ? )